Security/Security/ECC_impl.h

```diff
@@ -0,0 +1,97 @@
+#ifndef ECC_IMPL_H
+#define ECC_IMPL_H
+
+#include "ECC.h"
+#include "Logger.h"
+#include "Utils.h"
+using namespace std;
+
+// Function to encrypt data using ECC
+template<typename T>
+EncryptionResult ECC::encrypt(const T* data, const Point& publicKey) const {
+    // Check if the data is not null
+    if (data == nullptr) {
+        Logger::getInstance().log("encrypt: the data received is null");
+        throw std::runtime_error("The data received is null");
+    }
+    // Check if the public key is on the curve
+    if (!curve.isOnCurve(publicKey)) {
+        Logger::getInstance().log("encrypt: The public key is not correct - it is not on the curve");
+        throw std::runtime_error("The public key is not correct - it is not on the curve");
+    }
+
+    // Generate a random number k
+    cpp_int k = generateSecureRandomNumber(1, curve.getN());
+    // Compute the points R and S
+    Point R = curve.multiply(curve.getG(), k);
+    Point S = curve.multiply(publicKey, k);
+
+    // Convert data to a vector of unsigned char
```

Comment:
As we talked, no need to copy the data to another vector, you can use the pointer itself.